torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -403,6 +404,21 @@ def setUp(self) -> None:
         super().setUp()
         self._spawn_processes()
 
+    def _test_op_on_dtensor(self, dtensor: DTensor, op_call) -> None:
+        """
+        Unlike _test_op where the DTensor sharding is generated by DTensorConverter,
+        this function takes in DTensor object directly as argument and test the equality
+        of calling op on full_tensor() and DTensor.
+
+        Note: for in-place ops, the function may not work as expected if the DTensor
```

Comment:
shouldn't we be able to fix this limitation easily by just cloning the full tensor?

Reply:
I'm not very sure that this is an actual problem. But I guess you're right, I should just preventively do `full_tensor().detach().copy()`